components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2667,4 +2667,122 @@ void RewardsServiceImpl::GetAddressesForPaymentId(
                      callback));
 }
 
+uint64_t AutoContributionCountOnFileTaskRunner(
+    PublisherInfoDatabase* backend,
+    uint64_t reconcile_stamp) {
+  return backend->GetAutoContributeCount(reconcile_stamp);
+}
+
+void RewardsServiceImpl::GetAutoContributeCount(
+    const rewards_counter::OnRewardsCountedCallback callback) {
+  GetReconcileStamp(
+      base::Bind(&RewardsServiceImpl::GetCountByReconcileStamp,
+        AsWeakPtr(), callback));
+}
+
+uint64_t AutoContributionLastReconcileDateOnFileTaskRunner(
+    PublisherInfoDatabase* backend) {
+  return backend->GetLastReconcileDate();
+}
+
+void RewardsServiceImpl::GetLastReconcileStamp(
+    const rewards_counter::OnRewardsCountedCallback callback,
+    int64_t count) {
+  base::PostTaskAndReplyWithResult(
+      file_task_runner_.get(),
+      FROM_HERE,
+      base::Bind(&AutoContributionLastReconcileDateOnFileTaskRunner,
+          publisher_info_backend_.get()),
+      base::Bind(&RewardsServiceImpl::OnRewardsCounted,
+          AsWeakPtr(),
+          callback,
+          count));
+}
+
+void RewardsServiceImpl::GetCountByReconcileStamp(
+    const rewards_counter::OnRewardsCountedCallback callback,
+    uint64_t reconcile_stamp) {
+  base::PostTaskAndReplyWithResult(
+      file_task_runner_.get(),
+      FROM_HERE,
+      base::Bind(&AutoContributionCountOnFileTaskRunner,
+          publisher_info_backend_.get(), reconcile_stamp),
+      base::Bind(&RewardsServiceImpl::GetLastReconcileStamp,
+          AsWeakPtr(),
+          callback));
+}
+
+void RewardsServiceImpl::OnRewardsCounted(
+    rewards_counter::OnRewardsCountedCallback callback,
+    int64_t count,
+    uint64_t previous_reconcile_stamp) {
+  callback.Run(count, (previous_reconcile_stamp != 0 ? TimeFormatShortDate(
```

Comment:
please move this `if` outside of callback as it will be easier to read

Reply:
Fixed